### PR TITLE
More Python3 lint fixes

### PR DESCRIPTION
**Pull request by tbkka**
Some of the issues addressed include:
* Don't use `l` as a variable name (confusable with `1` or `I`)
* `print` statement does not exist in Py3, use `print` function instead
* Implicit tuple deconstruction in function args is no longer supported,
  use explicit splat `*` at the call site instead
* `xrange` does not exist in Py3, use `range` instead

### CODE REVIEW

**Comment by tbkka** [comment]
@swift-ci Python lint

**Comment by tbkka** [comment]
@swift-ci Please test

**Comment by tbkka** [comment]
There are only a couple of remaining Python3 lint issues in the tree.  I'll work through those after this gets merged.

File: utils/scale-test [hunk 1]
```diff
@@ -281,7 +281,7 @@ def Nelder_Mead_simplex(objective, params, bounds, epsilon=1.0e-6):
     locs = [tuple(random.uniform(*b) for b in bounds)
             for _ in range(ndim + 1)]
     SimplexPoint = namedtuple("SimplexPoint", ["loc", "val"])
-    simplex = [SimplexPoint(loc=l, val=f(l)) for l in locs]
+    simplex = [SimplexPoint(loc=L, val=f(L)) for L in locs]
```
Comment (compnerd) [inline utils/scale-test:284]: `loc` seems nicer than `l` or `L`.

File: utils/swift-api-dump.py [hunk 2]
```diff
@@ -161,8 +161,11 @@ def print_command(cmd, outfile=""):
 # Dump the API for the given module.
 
 
-def dump_module_api((cmd, extra_dump_args, output_dir, module, quiet,
-                     verbose)):
+def dump_module_api_star(pack):
```
Comment (compnerd) [inline utils/swift-api-dump.py:164]: Can't decide if this is better as `dump_module_api_packed` instead.

**Review by compnerd — APPROVED** [review]

File: utils/scale-test (hunk 1 above)
Comment (tbkka) [reply utils/scale-test:284]: I thought of that, but worried that `loc=loc` wouldn't read right.  If you think it makes sense that way, I'll change it.

File: utils/swift-api-dump.py (hunk 2 above)
Comment (tbkka) [reply utils/swift-api-dump.py:164]: `star` seems a meaningful term of art for anyone experienced with Python.  It's used in the standard library `starmap`, for instance.

**Comment by tbkka** [comment]
@swift-ci Please test

**Comment by tbkka** [comment]
The linux failure looks to be unrelated.  Trying again.

**Comment by tbkka** [comment]
@swift-ci Please test Linux

**Comment by tbkka** [comment]
@swift-ci Please test Linux

**Comment by tbkka** [comment]
@swift-ci Please smoke test

**Comment by tbkka** [comment]
@swift-ci Please smoke test Linux

**Review by CodaFi — APPROVED** [review]

**Pull request merged**